pkg/storaged/format-dialog.jsx

```diff
@@ -252,6 +250,26 @@ export function format_dialog(client, path, start, size, enable_dos_extended) {
     add_fsys(true, { value: "empty", title: _("No Filesystem") });
     add_fsys(true, { value: "custom", title: _("Custom (Enter filesystem type)") });
 
+    function add_crypto(storaged_name, title) {
+        if (client.manager.SupportedEncryptionTypes.indexOf(storaged_name) >= 0) {
+            let value = storaged_name;
+            if (storaged_name == client.manager.DefaultEncryptionType) {
+                title += " - " + _("Recommended default");
+                value = "default";
+            }
+            encryption_options.push({ value: value, title: title });
+        }
+    }
+
+    var encryption_options = [ ];
+    encryption_options.push({ value: "none", title: _("No encryption") });
+    if (client.manager.SupportedEncryptionTypes == undefined) {
+        encryption_options.push({ value: "default", title: _("LUKS") });
+    } else {
+        add_crypto("luks1", _("LUKS v1"));
+        add_crypto("luks2", _("LUKS v2"));
```

Comment:
Shouldn't this check the value of `SupportedEncryptionTypes` if LUKS v2 is actually available? E. g. on RHEL 7 we'd only have v1?

Reply:
`add_crypto` is doing the check.  Maybe we should call it `maybe_add_crypto`?